bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java

```diff
@@ -223,6 +240,10 @@ private void considerService(ServiceEvent serviceEvent) {
                         if (result != null) {
                             final DiscoveryResult resultNew = getLocalizedDiscoveryResult(result,
                                     FrameworkUtil.getBundle(participant.getClass()));
+                            final ServiceInfo service = serviceEvent.getInfo();
+                            if (participant.getRemovalGracePeriodSeconds(service) > 0) {
+                                cancelRemovalTask(service);
```

Comment:
Maybe this should be done unconditionally? If grace period in binding has changed to 0 since the task was scheduled, this might prevent the removal in case of any exception thrown.

Reply:
I don't think so. Most bindings (except the Miele one) will not have overridden the getRemoval method, so the default method is called and returns 0. So on those most bindings, we definitely must not differ from prior behavior. I don't want to break any other bindings.

Reply:
No harm done trying to remove it from the ConcurrentHashMap if it doesn't exist:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#remove-java.lang.Object-

Of course it doesn't change anything as long as no bindings supports this method. The PR itself doesn't change anything in this case. But when bindings will start supporting this, we can run into this situation where grace period will be changed from value >0 to 0 while having a task running. So I think this problem can be easily avoided by removing the condition?

Reply:
Ok.